src/core/geocms/qgsgeocmsconnection.h

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************
+    qgsgeocmsconnection.h
+    ---------------------
+    begin                : Feb 2017
+    copyright            : (C) 2017 by Muhammad Yarjuna Rohmat, Ismail Sunni
+    email                : rohmat at kartoza dot com, ismail at kartoza dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSGEOCMSCONNECTION_H
+#define QGSGEOCMSCONNECTION_H
+
+#define SIP_NO_FILE
+
+#include "qgis_core.h"
+#include "qgsdatasourceuri.h"
+#include "qgsmaplayer.h"
+#include "qgsproject.h"
+#include "qgsgeonoderequest.h"
+
+#include <QString>
+#include <QMultiMap>
+#include <QNetworkReply>
+
+
+/** \ingroup core
+ * Base class for all GeoCMS connection.
+ * Parent/children hierarchy is not based on QObject.
+*/
+class CORE_EXPORT QgsGeoCmsConnection : public QObject
```

Comment:
I don't think we need this kind of base class. Can we move its functionality to QgsGeoNodeConnection?